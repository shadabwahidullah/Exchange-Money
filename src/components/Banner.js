import React from 'react';

const HomeBanner = (props) => {
  const { title } = props;
  return (
    <div className="home-banner col-12 d-flex flex-column justify-content-center">
      <h1 className="fa align-self-center border border-warning border-3 rounded p-3 banner-title">{title}</h1>
    </div>
  );
};

export default HomeBanner;
