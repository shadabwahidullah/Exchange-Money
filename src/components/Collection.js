import React from 'react';
import { useDispatch } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { fetchCollection } from '../redux/reducers/collection';

const Collection = (props) => {
  const navigate = useNavigate();
  const { collectionInfo } = props;
  const {
    id,
    title,
    media_count: media,
    photos_count: photo,
    videos_count: video,
  } = collectionInfo;
  const dispatch = useDispatch();

  const callFetchCollection = (event) => {
    console.log('the event has been invoked: ', event);
    console.log('element parents id is ', event.target.parentElement.id);
    dispatch(fetchCollection(event.target.parentElement.id));
  };
  return (
    <div
      id={id}
      className="col-6 col-md-4 col-lg-3 col-xl-2 border border-white light-pink text-center collection-item d-flex flex-column justify-content-center"
    >
      <button
        type="button"
        onClick={(event) => {
          callFetchCollection(event);
          navigate('/targetCollection');
        }}
      >
        {title}
      </button>
      <span className="align-bottom">
        <p className="m-0">
          Media in total:
          {media}
        </p>
        <p className="m-0">
          Images:
          {photo}
          {'  '}
          Videos:
          {video}
        </p>
      </span>
    </div>
  );
};

export default Collection;
