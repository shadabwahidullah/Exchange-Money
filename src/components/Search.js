import React from 'react';
import { FaSearch } from 'react-icons/fa';

const Search = () => (
  <div>
    <FaSearch />
  </div>
);

export default Search;
