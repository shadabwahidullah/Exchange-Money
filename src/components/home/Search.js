import React from 'react';
import { useDispatch } from 'react-redux';
import { filterCollectionsAction } from '../../redux/reducers/collections';

const Search = () => {
  const dispatch = useDispatch();
  const filter = (event) => {
    dispatch(filterCollectionsAction(event.target.value));
  };

  return (
    <div className="row">
      <form>
        <input
          className="form-control m-2"
          placeholder="Search for images"
          onChange={(event) => {
            console.log('the typed in message is ', event.target.value);
            filter(event);
          }}
        />
      </form>
    </div>
  );
};
export default Search;
