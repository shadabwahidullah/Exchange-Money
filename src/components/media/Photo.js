import React from "react";

const Photo = () => {
    return (<di>Photo</di>)
}

export default Photo;