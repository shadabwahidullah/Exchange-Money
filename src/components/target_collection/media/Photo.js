import React from 'react';

const Photo = (props) => {
  const { photo } = props;
  const {
    id, photographer, src, height,
  } = photo;
  const { portrait } = src;

  return (
    <div className="col-6 p-0 pink border border-white border-3" id={id}>
      <img className="col-12" src={portrait} alt={photographer} />
      <h6 className="p-2 pb-0 fs-5">
        photo by:
        {' '}
        { photographer }
      </h6>
      <h6 className="p-2 m-0 fs-5">
        height:
        {' '}
        { height }
      </h6>
    </div>
  );
};

export default Photo;
