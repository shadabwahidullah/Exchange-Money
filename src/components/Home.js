import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import Collection from './Collection';
import { fetchCollections } from '../redux/reducers/collections';

const Home = () => {
  const collections = useSelector((state) => state.collectionsReducer);
  const dispatch = useDispatch();
  useEffect(() => {
    console.log('something is happening');
    dispatch(fetchCollections());
  }, []);
  return (
    <div className="col-12 row m-0 mt-5">
      {collections.Collections.map(
        (collection) => (
          <Collection key={collection.id} collectionInfo={collection}>
            collection
          </Collection>
        ),
      )}
    </div>
  );
};

export default Home;
