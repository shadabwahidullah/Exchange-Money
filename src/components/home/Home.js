import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Link, Outlet } from 'react-router-dom';
import Collection from '../Collection';
import { fetchCollections } from '../../redux/reducers/collections';
import HomeBanner from '../Banner';
import Search from './Search';

const Home = () => {
  const collections = useSelector((state) => state.collectionsReducer);
  const dispatch = useDispatch();
  useEffect(() => {
    console.log('something is happening');
    dispatch(fetchCollections());
  }, []);
  return (
    <div className="col-12 row m-0 mt-5">
      <HomeBanner title="Photo Collections Gallery" />
      <Search />
      {collections.Collections.map(
        (collection) => (
          <Link key={collection.id} to={`/target/${collection.id}`} className="col-6 col-md-4 col-lg-3 col-xl-2 p-0">
            <Collection key={collection.id} collectionInfo={collection}>
              collection
            </Collection>
          </Link>
        ),
      )}
      <Outlet />
    </div>
  );
};

export default Home;
