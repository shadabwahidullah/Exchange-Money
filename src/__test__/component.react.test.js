import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { Navbar } from 'react-bootstrap';
import store from '../redux/configureStore';
import Home from '../components/home/Home';
import Search from '../components/home/Search';
import Banner from '../components/Banner';
import Collection from '../components/Collection';
import Photo from '../components/target_collection/media/Photo';

describe('Components snapshot testing', () => {
  test('Home page testing', () => {
    const homePage = renderer.create(
      <Provider store={store}>
        <Home />
      </Provider>,
    ).toJSON;
    expect(homePage).toMatchSnapshot();
  });
  test('Search testing', () => {
    const search = renderer.create(
      <Provider store={store}>
        <Search />
      </Provider>,
    ).toJSON;
    expect(search).toMatchSnapshot();
  });
  test('Banner testing', () => {
    const banner = renderer.create(
      <Provider store={store}>
        <Banner />
      </Provider>,
    ).toJSON;
    expect(banner).toMatchSnapshot();
  });
  test('Collection testing', () => {
    const test = { id: 122, title: 'testing title', photos_count: '1239' };

    const collection = renderer.create(
      <Provider store={store}>
        <Collection collectionInfo={test} />
      </Provider>,
    ).toJSON;
    expect(collection).toMatchSnapshot();
  });
  test('Photo testing', () => {
    const testPhoto = {
      id: 'test',
      photographer: 'testPhotographer',
      src: { portrait: 'test source', medium: 'test source' },
      height: '200',
    };
    const photo = renderer.create(
      <Provider store={store}>
        <Photo photo={testPhoto} />
      </Provider>,
    ).toJSON;
    expect(photo).toMatchSnapshot();
  });
  test('Navbar testing', () => {
    const navbar = renderer.create(
      <Provider store={store}>
        <Navbar />
      </Provider>,
    ).toJSON;
    expect(navbar).toMatchSnapshot();
  });
});
