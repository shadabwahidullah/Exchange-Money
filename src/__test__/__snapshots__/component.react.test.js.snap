// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components snapshot testing Banner testing 1`] = `[Function]`;

exports[`Components snapshot testing Collection testing 1`] = `[Function]`;

exports[`Components snapshot testing Home page testing 1`] = `[Function]`;

exports[`Components snapshot testing Navbar testing 1`] = `[Function]`;

exports[`Components snapshot testing Photo testing 1`] = `[Function]`;

exports[`Components snapshot testing Search testing 1`] = `[Function]`;
